Python/Product/EnvironmentsList/PipExtension.xaml.cs

```diff
@@ -109,6 +109,16 @@ internal sealed partial class PipExtension : UserControl, ICanFocus {
                     }
                     Subcontext.DataContext = new PipEnvironmentView(view, _provider);
                 }
+            } else {
+                // Data context being deleted clean up old Pip info
+                // see ToolWindow.xaml.cs  _extensions.Clear();
+                var oldView = e.OldValue as EnvironmentView;
```

Comment:
Nitpick: This can be a little cleaner if you use the `is` keyword instead of `as`. This will attempt the cast and and avoid you having to check for null:

For example,

```csharp
if (e.OldValue is EnvironmentView oldView) {
    if (Subcontext.DataContext is PipEnvironmentView current) {
        current.Dispose();
    }
}
```